connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Constants.java

```diff
@@ -431,7 +433,8 @@ public class Constants {
     static SimpleAttributeDefinition TRACK_STATEMENTS = new SimpleAttributeDefinitionBuilder(TRACKSTATEMENTS_NAME, ModelType.STRING, true)
             .setAllowExpression(true)
             .setXmlName(Statement.Tag.TRACK_STATEMENTS.getLocalName())
-            .setDefaultValue(new ModelNode(Defaults.TRACK_STATEMENTS.name()))
+            .setDefaultValue(new ModelNode(Defaults.TRACK_STATEMENTS.toString()))
+            .setAllowedValues(Arrays.stream(Statement.TrackStatementsEnum.values()).map(Statement.TrackStatementsEnum::toString).toArray(String[]::new))
```

Comment:
@bstansberry as far as I can tell `setAlowedValues` doesn't actually enforce any validation so I'm not forcing the user to do anything (but that also means I didn't solve the problem)
I see the better way would be to write the values in lowercase but I'm guessing I need a custom marshaller for that?

Reply:
@michpetrov Sorry; it seems I misremembered! I think at one point someone investigated having this setting trigger automatic registration of a validator. But maybe it didn't happen for some reason or it just got recorded as an idea.

Anyway since this is the case, using setValidator(EnumValidator) seems better than setAllowedValues. DatasourceModelNodeUtil already calls Statement.TrackStatementsEnum.valueOf with this value and that will fail if the value is wrong. So the code is 'validating', just at the wrong point in operation execution.

EnumValidator also attempts to store the Enum.toString() value in the model, so using it results in accepting both the Enum.toString() and Enum.name() values, but storing toString().